lib/os/Kconfig

```diff
@@ -21,4 +21,23 @@ config BASE64
 	help
 	  Enable base64 encoding and decoding functionality
 
+config EVENTFD
+	bool "Enable support for eventfd"
+	depends on NET_BUF
+	help
+	  Enable support for event file descriptors, eventfd. An eventfd can
+	  be used as an event wait/notify mechanism together with POSIX calls
+	  like read, write and poll.
+
+if EVENTFD
+
+config EVENTFD_MAX
+	int "Maximum number of eventfd's"
+	default 1
+	range 1 4096
```

Comment:
Minor thing: Note that the `if EVENTFD` is equivalent to a `depends on EVENTFD`, and @ulfalizer will eventually clean it up to be so :)